pyxrf/model/setting.py

```diff
@@ -211,34 +335,114 @@ def get_roi_sum(self):
             nested dict as output
         """
         roi_result = {}
-        for fname, datav in six.iteritems(self.data_sets):
-            # quick way to ignore channel data, only for summed data
-            # to be updated
-            temp = {}
-            for k, v in six.iteritems(self.roi_dict):
-                leftv = v.left_val/1000
-                rightv = v.right_val/1000
-                sum2D = calculate_roi(datav.raw_data,
-                                      self.parameters['e_linear']['value'],
-                                      self.parameters['e_offset']['value'],
-                                      [leftv, rightv])
-                temp.update({k: sum2D})
-                logger.debug('Calculation is done for {}, {}, {}'.format(v.prefix,
-                                                                         fname, k))
-            roi_result[v.prefix+'_'+fname] = temp
-            return roi_result
-
-
-def calculate_roi(data3D, e_linear, e_offset, range_v):
+
+        datav = self.data_sets[self.data_title]
+
+        logger.info(f"Computing ROIs for dataset {self.data_title} ...")
+
+        temp = {}
+        data_raw = np.asarray(datav.raw_data)
+
+        if self.subtract_background:
+
+            logger.info(f"Subtracting background ...")
+
+            num_processors_to_use = multiprocessing.cpu_count()
+            logger.info(f"Cpu count: {format(num_processors_to_use)}")
```

Comment:
Is `format(...)` odd within the f-string?

Reply:
Yes

Reply:
Fixed